harness/determined/pytorch/deepspeed/dsat/_dsat_search_method.py

```diff
@@ -498,7 +499,8 @@ def on_trial_exited_early(
             logging.info(
                 "**** Shutting down DeepSpeed Autotune: Error in Model Profiling Info Trial ****"
             )
-            new_ops_list.append(searcher.Shutdown())
+            # Defer the shutdown to on_trial_closed() by simply closing the only trial
+            new_ops_list.append(searcher.Close(request_id))
```

Comment:
Confusingly, all trial which error our are already automatically closed, so having this explicit `Close` is unnecessary.  I would not be surprised if including such a `Close` would even lead to intermittent bugs of the kind we have previously hit.

I'm going to make a ticket to improve the Custom Searcher docs to clarify things like this.

Reply:
For clarities sake, the issue that came up when I didn't post that `Close` operation was that the test would hang, it would run out of events and not trigger the `should_shutdown` check properly